codec-http/src/main/java/io/netty/handler/codec/http/HttpContentEncoder.java

```diff
@@ -319,4 +327,22 @@ public EmbeddedChannel contentEncoder() {
             return contentEncoder;
         }
     }
+
+    private static final class RequestMetaData {
```

Comment:
@trustin To be clear: for HEAD / CONNECT we store the ZERO_LENGTH encoding? in isPassthru we still need to evaluate the response code, but that is available at processing time.

Can make the update.

@normanmaurer Not sure if Recycler would've worked in this situation because of the thread local. The request processing thread may not be the same as the response processing thread?